File: 0055-jump-game/0055-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        if(nums.size() == 1) return true;
        int far = 0;
        for(int i = 0; i < nums.size(); ++i) {
            far = max(far, nums[i] + i);
            if(i >= far && i != nums.size() - 1) return false;
            if(far >= nums.size() - 1) return true;
        }
        return false;
    }
};

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        if(n == 1) return true;
        int far = 0;
        
        for(int i = 0; i < n; ++i){
            far = Math.max(far, i + nums[i]);
            if(i >= far && i != n - 1) return false;
            if(far >= n - 1) return true;
        }
        
        return false;
    }
}